crawl-ref/source/mon-util.cc

```diff
@@ -3326,13 +3326,20 @@ static habitat_type _mons_class_habitat(monster_type mc,
     return ht;
 }
 
+habitat_type mons_habitat_types(monster_type t, monster_type base_t,
+                                bool real_amphibious) {
+    return _mons_class_habitat(fixup_zombie_type(t, base_t),
+                               real_amphibious);
```

Comment:
Can we add some test cases for this? In `catch2-tests/test_mon-util.cc`. It should be enough to test a few different monsters of different types including zombie types, simulacrum types, etc.

It would be best to also test `_describe_monster_hab()` separately (in a new `test_describe.cc` file): just testing one monster of each type (water, lava, none) is sufficient there.